enhancements/pipeline-tooling.md

```diff
@@ -0,0 +1,267 @@
+
+**Example 1:**
+
+`opm index merge quay.io/my/index1 quay.io/my/index2 -t quay.io/my/superindex -c=docker`
+
+**Example 2:**
+
+`opm index merge quay.io/my/index1 quay.io/my/index2 --generate -c=podman`
+
+### Get related images from an index
+
+As a pipeline admin, I want to know all the related images of a particular index so that I can know to update the index image when CVE fixes affect the related images of a bundle in my index.
+
+#### Design Proposal
+
+Add a flag to `opm index extract` called `related-images` that takes an index image and returns a list of related images from the database.
+
+**Example:**
+
+`opm index extract quay.io/my/index --related-images -c=docker`
+
+### CVE updates to running operators
+
+TODO(lance)
+
+### Design migration of upstream-community-operators
+
+As a maintainer of the upstream-community-operators git repository, I want my catalog build automation to use the new bundle/index format so that I can:
+
+- avoid storing operator manifests in the repository
+- simplify the addition, update, and removal of operators
+- reduce the time between an operator being submitted and it being displayed on operatorhub.io
+
+#### Design Proposal
+
+To design a migration, we'll paint a picture of a plausible final and work backward from there.
+
+On top of conformance with the rest of the ecosystem, the value-add of bundles and indexes for upstream-community-operators a reduction in operator manifests managed in the repository. Ideally, these new features will shrink the operator inclusion mechanism from a set of directories to a finite number of config files.
+
+Once fully migrated to the new formats, the general flow for updating the catalog's content might look like:
+
+1. The git repository is forked
+2. A bundle image reference is added, updated, or blacklisted in an index config file
+3. A PR is opened against the git repository
+4. An index is built with the patched config, using the current production index as a base
+5. The index content is verified and tested
+6. If tests pass, and the PR is approved, it enters a merge queue
+7. If the production index image changes before it merges, the PR drops from the queue (GOTO 4)
+8. Once merged, build a new production index image
+
+An configuration file will inform a build of the bundles to add or remove, if any, from an index:
+
+```yaml
+from: image://quay.io/operator-framework/community-operators:latest
+included:
+- ref: quay.io/coreos/something-bundle:v1
+  package: something
+  channel: stable
+- ref: quay.io/coreos/something-bundle@sha256:abcdef...
+  package: something
+  channel: alpha
+- ref: quay.io/redhat/something-else-bundle:v5.5
+  package: something-else
+  channel: v5.Y
+exluded:
+- ref: quay.io/redhat/something-else-bundle:v5.4
+  package: something-else
+  channel: v5.Y
+```
+
+A new `-f` option will be added to the `opm index add` subcommand which specifies the path of an index config file. The command will parse the given file and idempotently:
```

Comment:
I like this a lot, and it seems like it can be used for the CI to add a group of new bundles to an index (CVE goes out, for example).

Reply:
We could swing a little wider here and allow index references as well. This can be useful when, for example, I have separate indexes corresponding to the complete update graphs of different operators.